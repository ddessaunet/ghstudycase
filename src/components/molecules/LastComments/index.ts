export type { Props as LastCommentsProps } from './LastComments.d';
export { LastComments } from './LastComments';
