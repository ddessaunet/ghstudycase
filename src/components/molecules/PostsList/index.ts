export type { Props as PostsListProps } from './PostsList.d';
export { PostsList } from './PostsList';
