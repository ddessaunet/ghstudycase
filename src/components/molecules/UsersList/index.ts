export type { Props as UsersListProps } from './UsersList.d';
export { UsersList } from './UsersList';
