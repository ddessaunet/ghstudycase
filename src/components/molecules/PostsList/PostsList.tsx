import React from 'react';
import { Flex } from '@chakra-ui/react';
import { useNavigate } from 'react-router-dom';
import { Post } from 'components/atoms/Post';
import { LastComments } from 'components/molecules/LastComments';
import { getCommentsByPostPayload } from 'services/PostService';
import { Props } from './PostsList.d';

export const PostsList = ({ posts }: Props) => {
  const navigate = useNavigate();

  const nav = (path: string) => {
    window.scrollTo(0, 0);
    navigate(path);
  };

  return (
    <Flex flexDirection="column">
      {posts.map((post: any) => (
        <Flex key={post.id} m="10px">
          <Post
            {...post}
            action={(tagname: string) => nav(`/${tagname}/tag`)}
            request={() => getCommentsByPostPayload(post.id, 1, 0)}
          >
            <LastComments post={post} />
          </Post>
        </Flex>
      ))}
    </Flex>
  );
};
