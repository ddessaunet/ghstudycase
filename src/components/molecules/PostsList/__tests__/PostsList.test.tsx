import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { PostsList, PostsListProps } from 'components/molecules/PostsList';

describe('<PostsList />', () => {
  const props = {
    posts: [
      {
        id: '60d21b4667d0d8992e610c85',
        image: 'https://img.dummyapi.io/photo-1564694202779-bc908c327862.jpg',
        likes: 43,
        tags: ['animal', 'dog', 'golden retriever'],
        text: 'adult Labrador retriever',
        publishDate: '2020-05-24T14:53:17.598Z',
        owner: {
          id: '60d0fe4f5311236168a109ca',
          title: 'ms',
          firstName: 'Sara',
          lastName: 'Andersen',
          picture: 'https://randomuser.me/api/portraits/women/58.jpg',
        },
      },
    ],
  };
  const renderComponent = (props: PostsListProps) =>
    render(
      <MemoryRouter>
        <ChakraProvider>
          <PostsList {...props} />
        </ChakraProvider>
      </MemoryRouter>,
    );

  it('should render and match snapshot', () => {
    const component = renderComponent(props);
    expect(component).toMatchSnapshot();
  });
});
