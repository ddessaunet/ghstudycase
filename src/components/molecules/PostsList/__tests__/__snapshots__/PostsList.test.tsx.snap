// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsList /> should render and match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-light"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-j7qwjs"
      >
        <div
          class="css-cocn0m"
        >
          <div
            class="chakra-container css-140un8w"
          >
            <div
              class="css-j7qwjs"
            >
              <div
                class="css-obc5ee"
              >
                <div
                  class="css-1fd76z5"
                >
                  <img
                    alt=""
                    class="chakra-image css-1nbic85"
                    src="https://img.dummyapi.io/photo-1564694202779-bc908c327862.jpg"
                  />
                </div>
                <div
                  class="css-1dglqzs"
                >
                  <div
                    class="css-1o7kqr2"
                  >
                    <div
                      class="css-0"
                    >
                      <p
                        class="chakra-text css-ty7r4z"
                      >
                        2020-05-24T14:53:17.598Z
                      </p>
                    </div>
                    <div
                      class="css-0"
                    >
                      <p
                        class="chakra-text css-ty7r4z"
                      >
                        adult Labrador retriever
                      </p>
                    </div>
                    <div
                      class="css-2imjyh"
                    >
                      <a
                        class="chakra-link css-16aod2b"
                      >
                        animal
                      </a>
                      <a
                        class="chakra-link css-16aod2b"
                      >
                        dog
                      </a>
                      <a
                        class="chakra-link css-16aod2b"
                      >
                        golden retriever
                      </a>
                    </div>
                    <p
                      class="chakra-text css-1tx38fa"
                    >
                      Comments: 
                      Loading...
                    </p>
                  </div>
                  <div
                    class="css-zdpt2t"
                  >
                    <svg
                      width="20px"
                    >
                      like.svg
                    </svg>
                    <p
                      class="chakra-text css-1usi692"
                    >
                      43
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="css-j7qwjs"
              >
                <button
                  class="chakra-button css-1bi4nmq"
                  type="button"
                >
                  Show comments
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-j7qwjs"
    >
      <div
        class="css-cocn0m"
      >
        <div
          class="chakra-container css-140un8w"
        >
          <div
            class="css-j7qwjs"
          >
            <div
              class="css-obc5ee"
            >
              <div
                class="css-1fd76z5"
              >
                <img
                  alt=""
                  class="chakra-image css-1nbic85"
                  src="https://img.dummyapi.io/photo-1564694202779-bc908c327862.jpg"
                />
              </div>
              <div
                class="css-1dglqzs"
              >
                <div
                  class="css-1o7kqr2"
                >
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-ty7r4z"
                    >
                      2020-05-24T14:53:17.598Z
                    </p>
                  </div>
                  <div
                    class="css-0"
                  >
                    <p
                      class="chakra-text css-ty7r4z"
                    >
                      adult Labrador retriever
                    </p>
                  </div>
                  <div
                    class="css-2imjyh"
                  >
                    <a
                      class="chakra-link css-16aod2b"
                    >
                      animal
                    </a>
                    <a
                      class="chakra-link css-16aod2b"
                    >
                      dog
                    </a>
                    <a
                      class="chakra-link css-16aod2b"
                    >
                      golden retriever
                    </a>
                  </div>
                  <p
                    class="chakra-text css-1tx38fa"
                  >
                    Comments: 
                    Loading...
                  </p>
                </div>
                <div
                  class="css-zdpt2t"
                >
                  <svg
                    width="20px"
                  >
                    like.svg
                  </svg>
                  <p
                    class="chakra-text css-1usi692"
                  >
                    43
                  </p>
                </div>
              </div>
            </div>
            <div
              class="css-j7qwjs"
            >
              <button
                class="chakra-button css-1bi4nmq"
                type="button"
              >
                Show comments
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
